### PR TITLE
feat/header

### PR DESCRIPTION
**Pull request by zim89**
add packages: Sheet, Lucide
add Header
add SearchBar without functionality

### CODE REVIEW

**Comment by vlad-salone** [comment]
Check [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), there used in production

File: src/shared/ui/sheet.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+>(({ side = 'right', className, children, ...props }, ref) => (
+  <SheetPortal>
+    <SheetOverlay />
+    <SheetPrimitive.Content
+      ref={ref}
+      className={cn(sheetVariants({ side }), className)}
+      {...props}
+    >
+      {children}
+      {/* <SheetPrimitive.Close className='absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-secondary'>
```
Comment (vlad-salone) [inline src/shared/ui/sheet.tsx]: If no intend to use it, remove completely

File: package.json [hunk 2]
```diff
@@ -10,9 +10,11 @@
     "format": "prettier . --write"
   },
   "dependencies": {
+    "@radix-ui/react-dialog": "^1.0.5",
     "@radix-ui/react-slot": "^1.0.2",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.0",
+    "lucide-react": "^0.363.0",
```
Comment (vlad-salone) [inline package.json]: No need for `lucide-react` cause https://react-icons.github.io/react-icons/icons/lu/ already have it

File: src/widgets/header/ui/search-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'use client';
```
Comment (vlad-salone) [inline src/widgets/header/ui/search-bar.tsx]: This is a feature, also you can remove `'use client'` and done it with server actions

File: src/widgets/header/ui/navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+'use client';
```
Comment (vlad-salone) [inline src/widgets/header/ui/navbar.tsx:1]: navbar can be a separate feature with links props

File: src/widgets/header/ui/navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import { cn } from '~/shared/lib';
+
+const links = [
+  { label: 'Store', path: '/store' },
```
Comment (vlad-salone) [inline src/widgets/header/ui/navbar.tsx]: I assuming that this links are to sections except for contacts

File: src/widgets/header/ui/burger-menu.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    <Sheet>
+      <SheetTrigger className='xl:hidden'>
+        <AlignJustify className='stroke-1 transition-colors hover:text-black' />
+      </SheetTrigger>
+
+      <SheetContent
+        side='right'
+        className='w-full border-0 bg-card p-0 md:w-[23.5rem]'
+      >
+        {/* Header */}
```
Comment (vlad-salone) [inline src/widgets/header/ui/burger-menu.tsx]: Instead of comments you can separate this into components

**Review by vlad-salone — CHANGES_REQUESTED** [review]

File: src/widgets/header/ui/navbar.tsx (hunk 5 above)
Comment (zim89) [reply src/widgets/header/ui/navbar.tsx]: may be remove links to shared/const/nav.const.ts??? but i have eslint problems with inport

Comment (vlad-salone) [reply src/widgets/header/ui/navbar.tsx]: links must be at the level of page or app

File: src/shared/ui/sheet.tsx (hunk 1 above)
Comment (zim89) [reply src/shared/ui/sheet.tsx]: fixed

File: package.json (hunk 2 above)
Comment (zim89) [reply package.json]: remove lucide-react

File: src/widgets/header/ui/search-bar.tsx (hunk 3 above)
Comment (zim89) [reply src/widgets/header/ui/search-bar.tsx]: remove to features and remove 'use client'

File: src/widgets/header/ui/navbar.tsx (hunk 5 above)
Comment (zim89) [reply src/widgets/header/ui/navbar.tsx]: why should the **links** be on the `app` or `pages` layer?? there is a `const` segment in the `shared` layer that is available to all other layers

File: src/widgets/header/ui/burger-menu.tsx (hunk 6 above)
Comment (zim89) [reply src/widgets/header/ui/burger-menu.tsx]: remove comments

File: src/widgets/header/ui/navbar.tsx (hunk 5 above)
Comment (vlad-salone) [reply src/widgets/header/ui/navbar.tsx]: https://feature-sliced.design/docs/guides/issues/routes

**Review by vlad-salone — APPROVED** [review]

**Pull request merged**